client/app/pages/queries/hooks/useDeleteVisualization.js

```diff
@@ -0,0 +1,22 @@
+import { clone, extend, filter, isFunction } from "lodash";
+import { useRef, useCallback } from "react";
+import { Visualization } from "@/services/visualization";
+import notification from "@/services/notification";
+
+export default function useDeleteVisualization(query, onChange) {
+  const onChangeRef = useRef(null);
+  onChangeRef.current = isFunction(onChange) ? onChange : () => {};
+
+  return useCallback(
+    visualizationId =>
+      Visualization.delete({ id: visualizationId })
+        .$promise.then(() => {
+          const filteredVisualizations = filter(query.visualizations, v => v.id !== visualizationId);
+          onChangeRef.current(extend(clone(query), { visualizations: filteredVisualizations }));
```

Comment:
```suggestion
          onChangeRef.current(extend(query.clone(), { visualizations: filteredVisualizations }));
```

There's another in here too:
https://github.com/getredash/redash/blob/b20046593f7e0e04dbf2413c8915dd0cde47e554/client/app/components/queries/ApiKeyDialog/index.jsx#L25

Reply:
Fixed. Thank you!